app/jobs/create_gemini_recipient_ids_job.rb

```diff
@@ -8,24 +8,27 @@ class CreateGeminiRecipientIdsJob
   def perform(gemini_connection_id)
     gemini_connection = GeminiConnection.find(gemini_connection_id)
     # Users aren't able to create a recipient id if they are not fully verified
-    if gemini_connection.payable?
+    if gemini_connection.is_verified? && gemini_connection.status == "Active"
```

Comment:
Can we incorporate this check into a `payable_but_missing_recipient_id?` check and eliminate the details here?

Reply:
@tsmartt I've got several follow ups in this space for handling the race condition + notifying for oauth2 refresh failures.  This is an important UX issue so I vote we get it out and cleanup later.  @jlbyrne got to the heart of the issue and there is plenty of cleanup to be done afterward anyway.